shared/chat/conversation/messages/wrapper/container.js

```diff
@@ -1,140 +1,117 @@
+  _onShowMenu: (targetRect: ?ClientRect, message: Types.Message) =>
+    dispatch(
+      Route.navigateAppend([
+        {
+          props: {message, position: 'bottom left', targetRect},
+          selected: 'messageAction',
+        },
+      ])
+    ),
 })
 
-const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps: OwnProps) => {
-  const message = stateProps._message
-  const prevMessage = stateProps._prevMessage
-  const conversationState = stateProps._conversationState
-
-  const isEdited = message.type === 'Text' && stateProps._editedCount > 0
-  const isRevoked = !!message.senderDeviceRevokedAt
-  const failureDescription = message.messageState === 'failed' ? message.failureDescription : null
+const mergeProps = (stateProps, dispatchProps) => {
+  const {message, previous} = stateProps
 
-  const isFirstNewMessage = !!(
-    conversationState &&
-    message &&
-    message.messageID &&
-    conversationState.get('firstNewMessageID') === message.messageID
-  )
-
-  const skipMsgHeader = prevMessage && prevMessage.type === 'Text' && prevMessage.author === message.author
-
-  const firstMessageEver = !prevMessage
-  const firstVisibleMessage = prevMessage && Constants.messageKeyValue(prevMessage.key) === '1'
+  const continuingTextBlock =
+    previous &&
+    previous.author === message.author &&
+    (previous.type === 'text' || previous.type === 'deleted')
 
   const oldEnough =
-    prevMessage &&
-    prevMessage.timestamp &&
+    previous &&
+    previous.timestamp &&
     message.timestamp &&
-    message.timestamp - prevMessage.timestamp > Constants.howLongBetweenTimestampsMs
+    message.timestamp - previous.timestamp > howLongBetweenTimestampsMs
+
   const timestamp =
-    firstMessageEver || firstVisibleMessage || oldEnough ? formatTimeForMessages(message.timestamp) : null
-  const includeHeader = isFirstNewMessage || !skipMsgHeader || !!timestamp
+    stateProps.orangeLineAbove || !previous || oldEnough ? formatTimeForMessages(message.timestamp) : null
+  const includeHeader = !previous || !continuingTextBlock || !!timestamp
+  const failureDescription =
+    message.type === 'text' || message.type === 'attachment' ? message.errorReason : null
 
   return {
-    _editedCount: stateProps._editedCount,
-    _localMessageState: stateProps._localMessageState,
-    _message: stateProps._message,
-    _onAction: ownProps.onAction,
-    _onShowEditor: ownProps.onShowEditor,
-    _onUsernameClick: dispatchProps._onUsernameClick,
-    author: stateProps.author,
+    author: message.author,
     failureDescription,
+    hasOlderResetConversation: stateProps.hasOlderResetConversation,
     includeHeader,
-    innerClass: ownProps.innerClass,
+    innerClass: stateProps.innerClass,
     isBroken: stateProps.isBroken,
-    isEdited,
+    isEdited: message.hasBeenEdited,
     isEditing: stateProps.isEditing,
-    isFirstNewMessage,
+    isFirstNewMessage: false, //  TODO
```

Comment:
fixed in a future PR?

Reply:
yeah its actually removed. the orange line handling is done another way and this was just a leftover